docs/site/Serving-static-files.md

```diff
@@ -0,0 +1,64 @@
+---
+lang: en
+title: 'Serving static files in LoopBack 4'
+keywords: LoopBack
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Serving-static-files.html
+---
+
+One of the basic requirements of a web app is the ability to serve static files.
+Serving static files from a LoopBack is very simple - just call the
+`app.static(path, rootDir[, options])` method. The variables in the API are
+explained below.
+
+- `app`: An instance of a LoopBack application.
+- `path`: The path where the static assets are to be served from. Refer to
+  https://expressjs.com/en/4x/api.html#path-examples for possible values.
+- `rootDir`: The directory where the static assets are located on the file
+  system.
+- `options`: An otional object for configuring the underlying
+  [express.static](https://expressjs.com/en/4x/api.html#express.static).
+
+Here is an example of configuring an app to serve the static assets from a
+directory named `public` at `/`.
+
+```ts
+import {TodoListApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+const path = require('path');
+
+export async function main(options?: ApplicationConfig) {
+  const app = new TodoListApplication(options);
+  app.static('/', path.join(__dirname, 'public'));
```

Comment:
When reading this code, it's not clear to me whether `__dirname` is project root or `src`. 

AFAICT, the `main` function is exported by `src/index.ts`, which is compiled to `dist/src/index.js`. As a result, your example is mounting the directory `dist/src/public` as the static assets. I believe this is not going to work in most projects because static assets are not copied over by `tsc`.

Please fix the path to point to the `public` directory in the project root.

Also to make this easier to understand for readers, add a [Code caption](https://loopback.io/doc/en/contrib/code_samples.html#code-caption) to tell readers what file should contain this code snippet.

```
{% include code-caption.html content="src/index.ts" %}
```

---

Personally, I think it's the responsibility of the `Application` class to lead the static assets. Consider the following acceptance test: [examples/todo/test/acceptance/home-page.controller.acceptance.ts](https://github.com/strongloop/loopback-next/blob/81c85d3659b7009f4437ab2795379ac2e38db19f/examples/todo/test/acceptance/home-page.controller.acceptance.ts#L6-L16):

```ts
import {Client} from '@loopback/testlab';
import {TodoListApplication} from '../..';
import {setupApplication} from './test-helper';


describe('HomePageController', () => {
  let app: TodoListApplication;
  let client: Client;

  before('setupApplication', async () => {
    ({app, client} = await setupApplication());
  });

  // ...
});
```

Because the static assets are configured only in the `main` function, this acceptance test will not see the static assets and thus cannot verify that static assets were correctly configured.

My proposal:

Move the code setting up static assets to Application's constructor.

```ts
export class TodoListApplication extends BootMixin(
  ServiceMixin(RepositoryMixin(RestApplication)),
) {
  constructor(options: ApplicationConfig = {}) {
    super(options);

    // ...

    this.static('/', path.join(__dirname, '../../public'));
  }
}
```